prost-build/src/code_generator.rs

```diff
@@ -66,18 +63,12 @@ impl<'a> CodeGenerator<'a> {
             s
         });
 
-        let syntax = match file.syntax.as_ref().map(String::as_str) {
-            None | Some("proto2") => Syntax::Proto2,
-            Some("proto3") => Syntax::Proto3,
-            Some(s) => panic!("unknown syntax: {}", s),
-        };
-
         let mut code_gen = CodeGenerator {
             config,
             package: file.package.unwrap_or_default(),
             type_path: Vec::new(),
             source_info,
-            syntax,
+            syntax: file.syntax.as_ref().map(String::as_str).into(),
```

Comment:
When you change to `impl From<Option<&String>> for Syntax`, this could be simplified. Am I right?
```suggestion
            syntax: file.syntax.into(),
```

Reply:
I suppose it would. `&String` in method arguments is generally a bad practice, though.

Reply:
Well, `&String` is wrong anyway. This seems to work:
```suggestion
            syntax: file.syntax.as_deref().into(),
```

Reply:
Thanks!